roles/libvirt_manager/tasks/generate_networking_data.yml

```diff
@@ -0,0 +1,172 @@
+    _lnet_data: >-
+      {{
+        _network_data |
+        selectattr('name', 'match', _match) | first
+      }}
+    _networks: |
+      {{ _lnet_data.name | replace('cifmw_', '') }}:
+      {% if _lnet_data.ranges[0].start_v4 is defined and _lnet_data.ranges[0].start_v4 | length > 0                     %}
+        {% set net_4 = _lnet_data.ranges[0].start_v4 | ansible.utils.ipsubnet(_lnet_data.ranges[0].prefix_length_v4)    %}
+        network-v4: {{ net_4}}
+      {% endif                                                                                                          %}
+      {% if _lnet_data.ranges[0].start_v6 is defined and _lnet_data.ranges[0].start_v6 | length > 0                     %}
+        {% set net_6 = _lnet_data.ranges[0].start_v6 | ansible.utils.ipsubnet(_lnet_data.ranges[0].prefix_length_v6)  %}
+        network-v6: {{ net_6 }}
+      {% endif                                                                                                          %}
+    _gr_templates: |
+      {% for group in (_cifmw_libvirt_manager_layout.vms.keys() | sort) if group != 'controller' and
+                                                                           (_cifmw_libvirt_manager_layout.vms[group].amount | default(1) | int) > 0 %}
+      {% set _gr = (group == 'crc') | ternary('ocp', group)                                                             %}
+        {{ _gr }}s:
+          networks:
+            {{ _lnet_data.name | replace('cifmw_', '') }}:
+      {% if cifmw_networking_definition['group-templates'][_gr ~ 's']['network-template'] is not defined                %}
+              range: {{ cifmw_networking_definition['group-templates'][_gr ~ 's']['networks']['internalapi']['range'] }}
+      {% endif                                                                                                          %}
+      {% endfor                                                                                                         %}
+    _instances: |
+      {% if _cifmw_libvirt_manager_layout.vms.controller is defined %}
+      controller-0:
+        networks:
+          {{ _lnet_data.name | replace('cifmw_', '') }}:
+      {% if _lnet_data.ranges[0].start_v4 is defined and _lnet_data.ranges[0].start_v4 | length > 0 %}
+            ip-v4: "{{ _lnet_data.ranges[0].start_v4 | ansible.utils.ipsubnet(_lnet_data.ranges[0].prefix_length_v4) | ansible.utils.nthhost(9) }}"
+      {% endif %}
+      {% if _lnet_data.ranges[0].start_v6 is defined and _lnet_data.ranges[0].start_v6 | length > 0 %}
+            ip-v6: "{{ _lnet_data.ranges[0].start_v6 | ansible.utils.ipsubnet(_lnet_data.ranges[0].prefix_length_v6) | ansible.utils.nthhost(9) }}"
+      {% endif %}
+      {% endif %}
+  ansible.builtin.set_fact:
+    cifmw_networking_mapper_definition_patch_01_libvirt:
+      networks: "{{ _networks | from_yaml }}"
+      'group-templates': "{{ _gr_templates | from_yaml | default({}, true) }}"
```

Comment:
so here we may get an empty set for `group-templates` and `instances` - but I don't expect to get any for the `networks` one...